### PR TITLE
fix(core): Add keyboard support for mobile mode in menu

### PR DESCRIPTION
**Pull request by shrvr**
## Related Issue(s)

part of #8557 

## Description

Solves [this](https://github.com/SAP/fundamental-ngx/issues/8557#issuecomment-1220624224): Added  keyboard support for mobile mode in menu.

### CODE REVIEW

**Review by platon-rov — COMMENTED** [review]
two more items here

* when press "back" button being in 2nd level, you get to the 3rd level, press "back" again - you go somewhere, like you pressed on a link

https://user-images.githubusercontent.com/20265336/192984558-0dcbe080-0c70-4dc2-b4a6-74524c89af06.mov

* when you use right arrow key to get deeper to the 3rd level, then when you press left to get back to 2nd level - focus is lost

https://user-images.githubusercontent.com/20265336/192984824-76f595e1-5f13-417b-8b91-1d444bd43052.mov

File: libs/core/src/lib/menu/menu-mobile/menu-mobile.component.ts [hunk 1]
```diff
@@ -82,10 +82,11 @@ export class MenuMobileComponent extends MobileModeBase<MenuInterface> implement
 
     /** Navigate back to parent level of submenu */
     backToParentLevel(): void {
-        this._menuService.setActive(
-            false,
-            this._menuService.activeNodePath[this._menuService.activeNodePath.length - 1].item
-        );
+        const activeNode = this._menuService.activeNodePath[this._menuService.activeNodePath.length - 1].item;
+        this._menuService.setActive(false, activeNode);
+        setTimeout(() => {
```
Comment (mikerodonnell89) [inline libs/core/src/lib/menu/menu-mobile/menu-mobile.component.ts:87]: I do not think this setTimeout is needed (can just use `activeNode?.focus();`)

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]

**Comment by mikerodonnell89** [comment]
Using setTimeout here is giving off some code smells. It would be better to wait for the presence of the desired element, or to use events, something along those lines

File: libs/core/src/lib/menu/services/menu.service.ts [hunk 2]
```diff
@@ -256,7 +257,7 @@ export class MenuService {
         ) {
             if (this.focusedNode.parent?.item) {
                 this.setActive(false, this.focusedNode.parent.item);
-                this.setFocused(this.focusedNode.parent.item);
+                console.log('left arrow: focused node: ', this.focusedNode);
```
Comment (platon-rov) [inline libs/core/src/lib/menu/services/menu.service.ts:260]: please remove log.

**Comment by shrvr** [comment]
Current Issue (Help Needed for):

https://user-images.githubusercontent.com/65063487/197236672-432eb490-7220-4bc1-b594-4ce3ca677489.mp4

When we come back to Main Menu It focuses first item instead the item which we lastly visited.
In this case, it focused 'Fruits'  instead 'Milk Products' when we come back to main menu at last.

**Comment by mikerodonnell89** [comment]
Can we have a variable to store the previously focused item's index when moving between submenus?

**Comment by shrvr** [comment]
> Can we have a variable to store the previously focused item's index when moving between submenus?

Menu-service is keeping track of the `activepath` when we travel through submenus. There is already `focusedNode` that we can focus when we go backward. 
Issue is when we come back to main menu, it focuses first menu item. The `focus `function of menu-item is called one extra time when we come back to main menu.

**Review by mikerodonnell89 — APPROVED** [review]

**Pull request merged**